crates/web5/src/credentials/verifiable_credential_1_1.rs

```diff
@@ -316,51 +295,48 @@ impl VerifiableCredential {
 
         if let Some(id) = &vc_payload.id {
             if id != jti {
-                return Err(CredentialError::ClaimMismatch("id".to_string()));
+                return Err(CredentialError::ClaimMismatch("id".to_string()).into());
             }
         }
 
         if let Some(issuer) = &vc_payload.issuer {
             let vc_issuer = issuer.to_string();
             if iss != vc_issuer {
-                return Err(CredentialError::ClaimMismatch("issuer".to_string()));
+                return Err(CredentialError::ClaimMismatch("issuer".to_string()).into());
             }
         }
 
         if let Some(credential_subject) = &vc_payload.credential_subject {
             if sub != credential_subject.id {
-                return Err(CredentialError::ClaimMismatch("subject".to_string()));
+                return Err(CredentialError::ClaimMismatch("subject".to_string()).into());
             }
         }
 
-        let now = SystemTime::now();
-        match vc_payload.expiration_date {
-            Some(ref vc_payload_expiration_date) => match exp {
+        if let Some(vc_payload_expiration_date) = vc_payload.expiration_date {
+            match exp {
                 None => {
                     return Err(CredentialError::MisconfiguredExpirationDate(
                         "VC has expiration date but no exp in registered claims".to_string(),
-                    ));
+                    )
+                    .into());
                 }
                 Some(exp) => {
                     if vc_payload_expiration_date
-                        .duration_since(UNIX_EPOCH)?
+                        .duration_since(UNIX_EPOCH)
+                        .map_err(|e| {
+                            Web5Error::Unknown(format!("unknown system time error {}", e))
+                        })?
                         .as_secs()
-                        != exp.duration_since(UNIX_EPOCH)?.as_secs()
+                        != exp
+                            .duration_since(UNIX_EPOCH)
+                            .map_err(|e| {
+                                Web5Error::Unknown(format!("unknown system time error {}", e))
+                            })?
+                            .as_secs()
                     {
```

Comment:
this 11 line conditional is a little tough on the eyes. it runs its own conditionals and has two possible return statements all within the conditional part of the `if`

Reply:
simplified in latest commit. lmk whatcha think!